### PR TITLE
Order objects on API endpoints

### PR DESCRIPTION
**Pull request by dblenkus**

### CODE REVIEW

**Comment by kostko** [comment]
Tests are failing?

**Comment by mstajdohar** [comment]
> Tests are failing?

I think because of the problems with our JIRA/Jenkins server. Tests passed on Travis.

File: resolwe/flow/views.py [hunk 1]
```diff
@@ -451,10 +452,16 @@ class CollectionViewSet(ResolweCreateModelMixin,
                         viewsets.GenericViewSet):
     """API view for :class:`Collection` objects."""
 
-    queryset = Collection.objects.all().prefetch_related('descriptor_schema', 'contributor')
+    queryset = Collection.objects.all().prefetch_related(
+        'descriptor_schema',
+        'contributor',
+        Prefetch('data', queryset=Data.objects.all().order_by('created'))
```
Comment (mstajdohar) [inline resolwe/flow/views.py]: Created is not OK, we should order by ID. Objects may be created within the same microsecond and in this cases the order among them will be random.

File: resolwe/flow/views.py [hunk 2]
```diff
@@ -451,10 +452,16 @@ class CollectionViewSet(ResolweCreateModelMixin,
...
-    queryset = Collection.objects.all().prefetch_related('descriptor_schema', 'contributor')
+    queryset = Collection.objects.all().prefetch_related(
+        'descriptor_schema',
+        'contributor',
+        Prefetch('data', queryset=Data.objects.all().order_by('id'))
+    )
     serializer_class = CollectionSerializer
     permission_classes = (permissions_cls,)
     filter_class = CollectionFilter
+    ordering_fields = ('created', 'modified', 'name')
```
Comment (mstajdohar) [inline resolwe/flow/views.py]: Should we create db indexes on this fields?

File: resolwe/flow/views.py [hunk 3]
```diff
@@ -521,6 +530,8 @@ class DataViewSet(ResolweCreateDataModelMixin,
     serializer_class = DataSerializer
     permission_classes = (permissions_cls,)
     filter_class = DataFilter
+    ordering_fields = ('created', 'modified', 'started', 'finished', 'name')
```
Comment (mstajdohar) [inline resolwe/flow/views.py]: Should we create indexes on this fields?

**Comment by mstajdohar** [comment]
Fix tests.

**Comment by dblenkus** [comment]
I've just remove one test, because if we order by id, there is no way to change order and it cannot be tested if it works.

**Comment by mstajdohar** [comment]
> I've just remove one test, because if we order by id, there is no way to change order and it cannot be tested if it works.

:) 

You should test if the id field is of type auto incremental integer. If we change the type of IDs sometime in the future this sort will fail. But wait, then even your current test is valid.

**Comment by mstajdohar** [comment]
Tests still fail.

**Pull request merged**